### PR TITLE
chore(dropdowns): tag and document deprecated components

### PR DESCRIPTION
**Pull request by jzempel**
## Description

Add programmatic and demo pointers to the new `Combobox` component.

### CODE REVIEW

**Review by geotrev — APPROVED** [review]
Is there any way to test this out before merging, or will it be testable from `website.git` when the submodule is updated in a PR?

File: packages/dropdowns/demo/select.stories.mdx [hunk 1]
```diff
@@ -22,6 +22,10 @@ import README from '../README.md';
 
 # API
 
+**DEPRECATED:** use `@zendeskgarden/react-dropdowns.next`
+[Combobox](https://zendeskgarden.github.io/react-components/?path=/docs/packages-dropdowns-next-combobox)
+instead.
```
Comment (Francois-Esquire) [inline packages/dropdowns/demo/select.stories.mdx:27]: Should we be linking to the `Select` component (in `forms`) instead?

Comment (jzempel) [reply packages/dropdowns/demo/select.stories.mdx:27]: Nope. `Combobox` is the [replacement](https://garden.zendesk.com/components/combobox#select-only) for this custom `Select`. The `Select` in forms is a native `<select>` with no listbox styling.

**Comment by jzempel** [comment]
> Is there any way to test this out before merging

@geotrev I checked imports in the related `Story` files to ensure folks will see the intended deprecation message.

Screenshot 2023-07-24 at 3 11 24 PM

**Review by Francois-Esquire — APPROVED** [review]

**Pull request merged**